components/brave_ads/core/browser/search_result_ad/search_result_ad_converting_util.cc

```diff
@@ -193,155 +193,134 @@ bool SetSearchAdProperty(const schema_org::mojom::PropertyPtr& ad_property,
+                        found_attributes.begin(), found_attributes.end(),
+                        std::back_inserter(absent_attributes));
+
+    VLOG(2) << "Some of search result ad attributes were not specified: "
+            << base::JoinString(absent_attributes, ", ");
 
-    // Creatives has been parsed.
-    break;
+    return;
   }
 
-  return search_result_ads;
+  GURL target_url = search_result_ad->target_url;
+  search_result_ads->emplace(std::move(target_url),
+                             std::move(search_result_ad));
 }
 
-void LogSearchResultAdMap(const SearchResultAdMap& search_result_ads) {
-  if (!VLOG_IS_ON(1)) {
+void ConvertWebPageEntityProperty(
+    const schema_org::mojom::PropertyPtr& property,
+    SearchResultAdMap* search_result_ads) {
+  if (!property || property->name != kCreativesPropertyName) {
     return;
   }
 
-  if (search_result_ads.empty()) {
-    VLOG(1) << "Parsed search result ads list is empty.";
+  schema_org::mojom::ValuesPtr& values = property->values;
+  if (!values || !values->is_entity_values()) {
+    return;
+  }
+
+  for (const auto& ad_entity : values->get_entity_values()) {
+    ConvertSearchResultAdEntity(ad_entity, search_result_ads);
+  }
+}
+
+void LogSearchResultAdMap(const SearchResultAdMap& search_result_ads) {
+  if (!VLOG_IS_ON(2)) {
     return;
   }
 
-  VLOG(1) << "Parsed search result ads list:";
   for (const auto& search_result_ad_pair : search_result_ads) {
     const auto& search_result_ad = search_result_ad_pair.second;
-    VLOG(1) << "Ad with \"" << kDataPlacementId
-            << "\": " << search_result_ad->placement_id;
-    VLOG(1) << "  \"" << kDataCreativeInstanceId
-            << "\": " << search_result_ad->creative_instance_id;
-    VLOG(1) << "  \"" << kDataCreativeSetId
-            << "\": " << search_result_ad->creative_set_id;
-    VLOG(1) << "  \"" << kDataCampaignId
-            << "\": " << search_result_ad->campaign_id;
-    VLOG(1) << "  \"" << kDataAdvertiserId
-            << "\": " << search_result_ad->advertiser_id;
-    VLOG(1) << "  \"" << kDataLandingPage
-            << "\": " << search_result_ad->target_url;
-    VLOG(1) << "  \"" << kDataHeadlineText
-            << "\": " << search_result_ad->headline_text;
-    VLOG(1) << "  \"" << kDataDescription
-            << "\": " << search_result_ad->description;
-    VLOG(1) << "  \"" << kDataRewardsValue << "\": " << search_result_ad->value;
-    VLOG(1) << "  \"" << kDataConversionTypeValue
-            << "\": " << search_result_ad->conversion->type;
-    VLOG(1) << "  \"" << kDataConversionUrlPatternValue
-            << "\": " << search_result_ad->conversion->url_pattern;
-    VLOG(1) << "  \"" << kDataConversionAdvertiserPublicKeyValue
-            << "\": " << search_result_ad->conversion->advertiser_public_key;
-    VLOG(1) << "  \"" << kDataConversionObservationWindowValue
+    VLOG(2) << "Converted search result ad with \"" << kDataPlacementId
```

Comment:
Maybe we should increase this to a higher number to be less verbose?

Reply:
changed to VLOG(6)